### ISSUE TITLE
cursor.fetch leaks memory ? 

### ISSUE THREAD

**Issue by arturgspb** [plv8/plv8#191]
hi! 

My plv8 version is 1.5.3 (install in docker https://github.com/clkao/docker-postgres-plv8/blob/master/9.6-1.5/Dockerfile)
maybe related issue: https://github.com/plv8/plv8/issues/64

my_table - has 1 000 000 with 12 many types columns. No PK, Indexes, triggers and etc.
```
do $$
var plan = plv8.prepare( 'select v from my_table as v' );
var cursor = plan.cursor();
while (cursor.fetch()) {}
cursor.close();
plan.free();
plv8.elog(NOTICE, 'done');
$$ language plv8
```
My query eats + ~160mb each 3-5 seconds
Query with generate_series work without leaks.

Can it be anything other than a memory leak?

**Comment by JerrySievert** [comment]
it can be one of three things:

1) postgres itself leaking memory (doubtful because of how memory management works in pg)
2) v8 leaking (see https://github.com/plv8/plv8/blob/master/plv8.cc#L137-L177)
3) v8 holding onto memory until GC

the place that we would get a leak would have to be an internal v8 leak not calling `free` on the memory it allocates outside of pg's memory management.

does that make sense?

**Comment by arturgspb** [comment]
For example plpython3u has no that problem and plpy faster at 3 and more times.

```
do $$
inc = 0
for row in plpy.cursor( 'select 1 as inc, v from my_campaign as v' ):
    inc += row['inc']
plpy.info('done' + str(inc))
$$ language plpython3u
```
INFO:  done725386
CONTEXT:  PL/Python anonymous code block
Query returned successfully with no result in 11.7 secs.


```
do $$
var plan = plv8.prepare( 'select 1 as inc, v from my_campaign as v' );
var cursor = plan.cursor();
var row;
inc = 0;
while (row = cursor.fetch()) {
    inc += row['inc']
}
cursor.close();
plan.free();
plv8.elog(NOTICE, 'done' + inc);
$$ language plv8
```
NOTICE:  done725386
Query returned successfully with no result in 01:05 minutes.


Maybe I'm doing something else, and you can optimize my code?

related code: https://github.com/postgres/postgres/blob/master/src/pl/plpython/plpy_cursorobject.c

**Comment by eugwne** [comment]
It might leak because of possibly missed SPI_freetuptable(SPI_tuptable); If you fetch 50 rows at a time that should be much faster than fetch one by one.

**Comment by arturgspb** [comment]
@eugwne, I'm not sure I understand you. I can do anything with my code to fix the problem?

**Comment by eugwne** [comment]
leak is strange, I thought it was fixed here #64 .... to make it faster you can call cursor.fetch(number) if I'm not mistaken
